src/debug/panic.c

```diff
@@ -88,5 +88,9 @@ void __panic(uint32_t p, char *filename, uint32_t linenum)
 				 filename, strlen + 1));
 	}
 
+	/* To distinguish regular panic() calls from exceptions */
+	__asm__ __volatile__("movi a3, 0\n\twsr.exccause a3\n\tesync" : : :
+			     "a3", "memory");
+
```

Comment:
I think a better comment. So, it is not clear for me why are we setting a3 to 0? This should be somehow clear from the comment.

Reply:
Updated.